lib/ClangImporter/ClangImporter.cpp

```diff
@@ -2432,6 +2432,84 @@ static bool isVisibleClangEntry(clang::ASTContext &ctx,
   return true;
 }
 
+TypeDecl *
+ClangModuleUnit::lookupNestedType(Identifier name,
+                                  const NominalTypeDecl *baseType) const {
+  // Special case for error code enums: try looking directly into the struct
+  // first. But only if it looks like a synthesized error wrapped struct.
+  if (name == getASTContext().Id_Code && !baseType->hasClangNode() &&
+      isa<StructDecl>(baseType) && !baseType->hasLazyMembers() &&
+      baseType->isChildContextOf(this)) {
+    auto *mutableBase = const_cast<NominalTypeDecl *>(baseType);
+    auto codeEnum = mutableBase->lookupDirect(name,/*ignoreNewExtensions*/true);
+    if (codeEnum.size() == 1 && isa<TypeDecl>(codeEnum.front()))
+      return cast<TypeDecl>(codeEnum.front());
+    if (codeEnum.size() > 1)
```

Comment:
Ah, so this is about how we import error enums, which come out something like this:

```swift
struct MyError : _BridgedStoredNSError {
  static let _domain: String
  var rawValue: NSError

  @objc enum Code: Int {
    case bad
    case worse
  }

  static var bad: MyError { return NSError(domain: _domain, code: Code.bad.rawValue) }
  static var worse: MyError { return NSError(domain: _domain, code: Code.worse.rawValue) }
}
```

However, it's possible we've just imported a struct that *looks* like an error enum struct (given the imperfect checks above), and that it's been extended with members named `Code`, possibly more than one. In that case we can't be sure this lookup is doing the right thing, and so give up on the "fast path".